src/core/ui/components/sign-in/magic-link-verify-email.tsx

```diff
@@ -0,0 +1,80 @@
+import { useCallback, useMemo, useState } from 'react';
```

Comment:
This component seems really similar to the magic link component? Any particular reason you chose to copy/pasta rather than reuse?